Fix PR creation gap after doctor test-fix loop

## Problem

When the builder writes code but doesn't commit/create a PR, and the doctor test-fix loop succeeds at fixing tests, the shepherd still fails with:

```
[OK] Tests now pass after Doctor fixes (54s)
[OK] Builder phase complete - PR #None created (345s)
[ERROR] Cannot enter Judge phase: no PR was created during Builder phase
```

This happened on issue #1898 — the builder created all the code, the doctor fixed formatting issues, tests passed, but nobody ever committed or created the PR. The shepherd exited with code 1 and required manual intervention to commit, push, and create the PR.

## Root Cause

The doctor test-fix loop in `cli.py` (lines ~424-524) re-verifies tests after the doctor runs, but the code path that returns to the main flow doesn't check whether a PR actually exists. The completion retry logic in `builder.py` (lines ~251-298) exists but may not trigger after the doctor test-fix path — the flow appears to be:

1. Builder runs → creates code but doesn't commit
2. Test verification fails (new errors detected)
3. Doctor test-fix runs → fixes formatting/test issues
4. Re-verification passes
5. Builder phase reports success with `PR #None`
6. Judge phase immediately fails — no PR exists

The completion retry (`_run_completion_phase`) only runs inside `_validate_and_complete()` which is called before test verification, not after the doctor loop.

## Proposed Fix

After the doctor test-fix loop succeeds (tests pass), before returning success from the builder phase:

1. Check if a PR exists for the issue
2. If not, run the same diagnostics as `_gather_incomplete_diagnostics()`:
   - Are there uncommitted changes in the worktree?
   - Are there commits ahead of main that haven't been pushed?
   - Does the remote branch exist?
3. If incomplete work is detected, run `_run_completion_phase()` to commit, push, and create the PR
4. Only then return success

This is essentially wiring the completion retry into the post-doctor-fix path.

## Files to Change

- `loom-tools/src/loom_tools/shepherd/cli.py` — after doctor test-fix success, call completion validation
- `loom-tools/src/loom_tools/shepherd/phases/builder.py` — may need to expose `_validate_and_complete()` or a subset for reuse

## Test Plan

- Unit test: mock builder that creates files but doesn't commit, doctor that fixes tests → verify completion phase triggers
- Integration: run shepherd on an issue where builder leaves uncommitted work → verify PR gets created after doctor fixes

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -472,13 +472,35 @@ def orchestrate(ctx: ShepherdContext) -> int:
                         f"Doctor determined test failures are pre-existing ({doctor_elapsed}s)"
                     )
                     completed_phases.append("Doctor (pre-existing failures)")
+
+                    # Run validation and completion to ensure PR exists
+                    # Even with pre-existing failures, we need to verify the
+                    # builder actually committed and created a PR
+                    _print_phase_header("PHASE 3d: COMPLETION VALIDATION (pre-existing failures)")
+                    completion_start = time.time()
+                    completion_result = builder.validate_and_complete(ctx)
+                    completion_elapsed = int(time.time() - completion_start)
+                    builder_total_elapsed += completion_elapsed
+
+                    if completion_result.is_shutdown:
+                        raise ShutdownSignal(completion_result.message)
+
+                    if completion_result.status == PhaseStatus.FAILED:
+                        log_error(completion_result.message)
+                        _mark_builder_no_pr(ctx)
+                        return 1
+
                     # Continue to PR creation - pre-existing failures are acceptable
                     result = PhaseResult(
                         status=PhaseStatus.SUCCESS,
                         message="builder complete (pre-existing test failures)",
                         phase_name="builder",
-                        data={"preexisting_failures": True},
+                        data={
+                            "preexisting_failures": True,
+                            "pr_number": completion_result.data.get("pr_number"),
+                        },
                     )
+                    log_success(f"Completion validation passed ({completion_elapsed}s)")
                     break
 
                 if doctor_result.status in (PhaseStatus.FAILED, PhaseStatus.STUCK):
@@ -507,14 +529,37 @@ def orchestrate(ctx: ShepherdContext) -> int:
                 builder_total_elapsed += test_elapsed
 
                 if test_result is None:
-                    # Tests passed
+                    # Tests passed - now validate PR exists and complete if needed
                     log_success(f"Tests now pass after Doctor fixes ({test_elapsed}s)")
+
+                    # Run validation and completion to ensure PR exists
+                    # This handles the case where builder created code but
+                    # didn't commit/push/create PR before doctor fixed tests
+                    _print_phase_header("PHASE 3d: COMPLETION VALIDATION (after Doctor fixes)")
+                    completion_start = time.time()
+                    completion_result = builder.validate_and_complete(ctx)
+                    completion_elapsed = int(time.time() - completion_start)
+                    builder_total_elapsed += completion_elapsed
+
+                    if completion_result.is_shutdown:
+                        raise ShutdownSignal(completion_result.message)
+
+                    if completion_result.status == PhaseStatus.FAILED:
+                        log_error(completion_result.message)
+                        _mark_builder_no_pr(ctx)
+                        return 1
+
+                    # Use the completion result which has the PR number
                     result = PhaseResult(
                         status=PhaseStatus.SUCCESS,
                         message="builder complete (tests fixed by Doctor)",
                         phase_name="builder",
-                        data={"test_fixed_by_doctor": True},
+                        data={
+                            "test_fixed_by_doctor": True,
+                            "pr_number": completion_result.data.get("pr_number"),
+                        },
                     )
+                    log_success(f"Completion validation passed ({completion_elapsed}s)")
                     break
                 else:
                     # Tests still failing - update result and loop

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -369,6 +369,114 @@ class BuilderPhase:
         """
         return self._run_test_verification(ctx)
 
+    def validate_and_complete(self, ctx: ShepherdContext) -> PhaseResult:
+        """Validate builder phase and run completion if needed.
+
+        This method is used after the doctor test-fix loop succeeds to ensure
+        that a PR actually exists. If the builder made changes but didn't
+        complete the commit/push/PR workflow, this will spawn a focused
+        completion phase to finish the work.
+
+        This is the same validation and completion logic that runs at the end
+        of the normal builder.run() method, extracted for reuse.
+
+        Returns:
+            PhaseResult with SUCCESS status if PR exists or was created.
+            PhaseResult with FAILED status if completion could not be achieved.
+        """
+        # Run validation with completion retry
+        completion_attempts = 0
+        max_completion_attempts = ctx.config.builder_completion_retries
+
+        while True:
+            if self.validate(ctx):
+                break  # Validation passed
+
+            # Re-diagnose on each iteration to detect partial progress
+            diag = self._gather_diagnostics(ctx)
+
+            # Check if this is incomplete work that could be completed
+            if not self._has_incomplete_work(diag):
+                # No incomplete work pattern — nothing to retry
+                return PhaseResult(
+                    status=PhaseStatus.FAILED,
+                    message=(
+                        f"builder phase validation failed after doctor fixes: "
+                        f"{diag['summary']}"
+                    ),
+                    phase_name="builder",
+                    data={"diagnostics": diag, "completion_attempts": completion_attempts},
+                )
+
+            if completion_attempts >= max_completion_attempts:
+                # Retries exhausted — try direct fallback for mechanical ops
+                if self._direct_completion(ctx, diag):
+                    log_success("Direct completion succeeded after doctor fixes")
+                    continue  # Re-validate
+
+                return PhaseResult(
+                    status=PhaseStatus.FAILED,
+                    message=(
+                        f"builder phase validation failed after doctor fixes: "
+                        f"{diag['summary']}"
+                    ),
+                    phase_name="builder",
+                    data={"diagnostics": diag, "completion_attempts": completion_attempts},
+                )
+
+            completion_attempts += 1
+            log_warning(
+                f"Builder left incomplete work after doctor fixes "
+                f"(attempt {completion_attempts}/{max_completion_attempts}): "
+                f"{diag['summary']}"
+            )
+
+            # Run completion phase with attempt number for progressive simplification
+            completion_exit = self._run_completion_phase(
+                ctx, diag, attempt=completion_attempts
+            )
+
+            if completion_exit == 3:
+                # Shutdown during completion
+                return PhaseResult(
+                    status=PhaseStatus.SHUTDOWN,
+                    message="shutdown signal detected during completion phase",
+                    phase_name="builder",
+                )
+
+            if completion_exit == 0:
+                log_info("Completion phase finished, re-validating")
+                continue  # Re-validate
+
+            log_warning(f"Completion phase failed with exit code {completion_exit}")
+            # Loop back to re-diagnose and potentially retry
+
+        # Get PR number
+        pr = get_pr_for_issue(ctx.config.issue, repo_root=ctx.repo_root)
+        if pr is None:
+            diag = self._gather_diagnostics(ctx)
+            return PhaseResult(
+                status=PhaseStatus.FAILED,
+                message=(
+                    f"could not find PR for issue #{ctx.config.issue} after doctor fixes: "
+                    f"{diag['summary']}"
+                ),
+                phase_name="builder",
+                data={"diagnostics": diag},
+            )
+
+        ctx.pr_number = pr
+
+        # Report PR created
+        ctx.report_milestone("pr_created", pr_number=pr)
+
+        return PhaseResult(
+            status=PhaseStatus.SUCCESS,
+            message=f"builder phase complete after doctor fixes - PR #{pr} created",
+            phase_name="builder",
+            data={"pr_number": pr},
+        )
+
     def _fetch_issue_body(self, ctx: ShepherdContext) -> str | None:
         """Fetch the issue body from GitHub.
 

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -5873,3 +5873,219 @@ class TestBuilderSupplementalVerification:
         assert result is not None
         assert result.status == PhaseStatus.FAILED
         assert "supplemental" in result.message
+
+
+class TestBuilderValidateAndComplete:
+    """Tests for BuilderPhase.validate_and_complete() method.
+
+    This method is used after the doctor test-fix loop succeeds to ensure
+    that a PR actually exists. It runs the same validation and completion
+    logic as the end of builder.run().
+    """
+
+    def test_validate_and_complete_success_when_pr_exists(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should return SUCCESS when PR exists."""
+        builder = BuilderPhase()
+
+        with (
+            patch.object(builder, "validate", return_value=True),
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue",
+                return_value=123,
+            ),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.SUCCESS
+        assert "PR #123" in result.message
+        assert result.data.get("pr_number") == 123
+        assert mock_context.pr_number == 123
+
+    def test_validate_and_complete_runs_completion_when_incomplete(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should run completion phase when work is incomplete."""
+        builder = BuilderPhase()
+        mock_context.config.builder_completion_retries = 2
+
+        # First validation fails, second succeeds (after completion)
+        validate_calls = [False, True]
+
+        incomplete_diag = {
+            "summary": "has uncommitted changes",
+            "worktree_exists": True,
+            "has_uncommitted_changes": True,
+            "commits_ahead": 0,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", side_effect=validate_calls),
+            patch.object(builder, "_gather_diagnostics", return_value=incomplete_diag),
+            patch.object(builder, "_has_incomplete_work", return_value=True),
+            patch.object(builder, "_run_completion_phase", return_value=0),
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue",
+                return_value=456,
+            ),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.SUCCESS
+        assert "PR #456" in result.message
+        assert result.data.get("pr_number") == 456
+
+    def test_validate_and_complete_fails_when_no_incomplete_work(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should fail when validation fails and no incomplete work."""
+        builder = BuilderPhase()
+
+        no_work_diag = {
+            "summary": "worktree does not exist",
+            "worktree_exists": False,
+            "has_uncommitted_changes": False,
+            "commits_ahead": 0,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", return_value=False),
+            patch.object(builder, "_gather_diagnostics", return_value=no_work_diag),
+            patch.object(builder, "_has_incomplete_work", return_value=False),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert "builder phase validation failed after doctor fixes" in result.message
+        assert result.data.get("diagnostics") == no_work_diag
+
+    def test_validate_and_complete_exhausts_retries(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should fail after exhausting completion retries."""
+        builder = BuilderPhase()
+        mock_context.config.builder_completion_retries = 2
+
+        incomplete_diag = {
+            "summary": "has uncommitted changes but completion keeps failing",
+            "worktree_exists": True,
+            "has_uncommitted_changes": True,
+            "commits_ahead": 0,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", return_value=False),
+            patch.object(builder, "_gather_diagnostics", return_value=incomplete_diag),
+            patch.object(builder, "_has_incomplete_work", return_value=True),
+            patch.object(builder, "_run_completion_phase", return_value=1),  # Always fail
+            patch.object(builder, "_direct_completion", return_value=False),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert "validation failed after doctor fixes" in result.message
+        assert result.data.get("completion_attempts") == 2
+
+    def test_validate_and_complete_handles_shutdown(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should return SHUTDOWN when completion phase shuts down."""
+        builder = BuilderPhase()
+        mock_context.config.builder_completion_retries = 2
+
+        incomplete_diag = {
+            "summary": "has uncommitted changes",
+            "worktree_exists": True,
+            "has_uncommitted_changes": True,
+            "commits_ahead": 0,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", return_value=False),
+            patch.object(builder, "_gather_diagnostics", return_value=incomplete_diag),
+            patch.object(builder, "_has_incomplete_work", return_value=True),
+            patch.object(builder, "_run_completion_phase", return_value=3),  # Shutdown
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.SHUTDOWN
+        assert "shutdown" in result.message.lower()
+
+    def test_validate_and_complete_tries_direct_completion_as_fallback(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should try direct completion when retries exhausted."""
+        builder = BuilderPhase()
+        mock_context.config.builder_completion_retries = 1
+
+        # First two validate calls fail, third succeeds (after direct completion)
+        validate_calls = [False, False, True]
+
+        incomplete_diag = {
+            "summary": "has commits but needs push",
+            "worktree_exists": True,
+            "has_uncommitted_changes": False,
+            "commits_ahead": 1,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", side_effect=validate_calls),
+            patch.object(builder, "_gather_diagnostics", return_value=incomplete_diag),
+            patch.object(builder, "_has_incomplete_work", return_value=True),
+            patch.object(builder, "_run_completion_phase", return_value=1),  # Fail
+            patch.object(builder, "_direct_completion", return_value=True),  # Succeed
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue",
+                return_value=789,
+            ),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.SUCCESS
+        assert "PR #789" in result.message
+
+    def test_validate_and_complete_fails_when_pr_not_found_after_validation(
+        self, mock_context: MagicMock
+    ) -> None:
+        """validate_and_complete should fail if PR not found even after validation passes."""
+        builder = BuilderPhase()
+
+        fake_diag = {
+            "summary": "worktree exists but no PR",
+            "worktree_exists": True,
+            "has_uncommitted_changes": False,
+            "commits_ahead": 0,
+            "remote_branch_exists": True,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch.object(builder, "validate", return_value=True),
+            patch.object(builder, "_gather_diagnostics", return_value=fake_diag),
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue",
+                return_value=None,  # PR still not found
+            ),
+        ):
+            result = builder.validate_and_complete(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert "could not find PR" in result.message
+        assert "after doctor fixes" in result.message

